### PR TITLE
#910 Support multiple similar payment methods

### LINKED ISSUE #910: Cannot add more than one similar payment method.
### Problem

Go to "Income Details", add multiple "Other" payments and click save.
Screenshot 2020-05-02 at 11 35 35

Re-open Income Details, only the last "Other" payment was saved.

Screenshot 2020-05-02 at 11 36 19

### Solution

Allow similar payment methods to be stored.

Currently, the payments info are saved in an object where:

```js
{
   other: {
     value: 'mbway: 912345678'
  }
}
```

Because an object cannot have multiple similar keys (e.g. 3 "other"), only the last payment key will be stored.

The payments must be saved in an array:

```js
[
  {
     name: 'Other',
     value: 'mbway: 912345678'
  }
]
```

### PR DESCRIPTION
**Pull request by sandrina-p**
Fixes #910

Note: When PR #884 is merged (after or before this PR), the `ProfileCard.vue` needs to be updated to fetch correctly `ourGroupProfile.paymentMethods`.

### CODE REVIEW

File: frontend/model/contracts/group.js [hunk 1]
```diff
@@ -536,17 +536,17 @@ DefineContract({
...
-          venmo: objectMaybeOf({ value: string }),
-          other: objectMaybeOf({ value: string })
-        }),
         nonMonetaryEdit: objectOf({
           replace: string,
           with: string
         }),
-        nonMonetaryRemove: string
+        nonMonetaryRemove: string,
+        paymentMethods: optional(arrayOf(
```
Comment (taoeffect) [inline frontend/model/contracts/group.js]: Since we're already inside of an `objectMaybeOf` the `optional` isn't needed here.

**Review by taoeffect — CHANGES_REQUESTED** [review]
Excellent! Only one issue (very minor) noticed!

**Review by taoeffect — DISMISSED** [review]
Approved! Excellent work! 😄

File: frontend/model/contracts/group.js (hunk 1 above)
Comment (sandrina-p) [reply frontend/model/contracts/group.js]: Before it had `objectMaybeOf`. That's why I kept the `optional` thing. Removed it.

**Review by taoeffect — APPROVED** [review]
Approved again since my previous review was cleared and tests now pass!

**Pull request merged**